python-package/setup.py

```diff
@@ -35,7 +36,13 @@ def copy_files_helper(folder_name):
         src = os.path.join(CURRENT_DIR, os.path.pardir, folder_name)
         if os.path.exists(src):
             dst = os.path.join(CURRENT_DIR, 'compile', folder_name)
-            shutil.rmtree(dst, ignore_errors=True)
+            if os.path.exists(dst):
+                if os.path.isdir:
+                    # see https://github.com/pypa/distutils/pull/21
```

Comment:
Should other calls to `shutil.rmtree` be replaced too?
https://github.com/microsoft/LightGBM/blob/b360700411a31085af2319294a5553e7c667a242/python-package/setup.py#L76
https://github.com/microsoft/LightGBM/blob/b360700411a31085af2319294a5553e7c667a242/python-package/setup.py#L101
https://github.com/microsoft/LightGBM/blob/b360700411a31085af2319294a5553e7c667a242/python-package/setup.py#L260
https://github.com/microsoft/LightGBM/blob/b360700411a31085af2319294a5553e7c667a242/python-package/setup.py#L262